### DISCUSSION TITLE
Limit Reset Password/Forgot Password with attempts limit or recaptcha

### DISCUSSION THREAD

**Discussion by tzahifadida** [keycloak/keycloak#15871] [Q&A]
Hi,
Is there a way to protect the "Forgot Password" from spamming my users. Currently, a malicious party can simply hit forgot password and enter an email and it is sent without any limits. Moreover, if my users will get spammed the email address will be marked as a spam and therefor move to the spam folder.

I noticed there is a Recaptcha step in the registration form, but I could not add it in the "reset credentials" in the "Authentication" section. Is there a way to do that?

Would love to solve that.
Thanks.

**Comment by davidezordan25** [comment]
Still not resolved in Keycloak 25.0.1

**Comment by lamoboos223** [comment]
you can create custom extension using the `Authentication SPI` where you call recaptcha and also update the themes to add recaptcha html code

@tzahifadida
